apps/dashboard/src/@/components/blocks/Img.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+/* eslint-disable @next/next/no-img-element */
+
+import { useState } from "react";
+import { cn } from "../../lib/utils";
+
+type imgElementProps = React.DetailedHTMLProps<
+  React.ImgHTMLAttributes<HTMLImageElement>,
+  HTMLImageElement
+> & {
+  skeleton?: React.ReactNode;
+  fallback?: React.ReactNode;
+};
+
+export function Img(props: imgElementProps) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const { className, ...restProps } = props;
+
+  const showSkeleton = isLoading || !props.src;
+  const defaultSkeleton = <div className="animate-pulse bg-accent" />;
+  const defaultFallback = <div className="bg-muted" />;
+
+  return (
+    <div className="relative">
+      {/* biome-ignore lint/a11y/useAltText: <explanation> */}
+      <img
+        {...restProps}
+        onLoad={() => {
+          setIsLoading(false);
+        }}
+        onError={() => {
+          setError(true);
+        }}
+        style={{
+          opacity: showSkeleton ? 0 : 1,
+        }}
+        className={cn(
+          "fade-in-0 object-cover transition-opacity duration-300",
+          className,
+        )}
+      />
+      {showSkeleton && (
+        <div
+          className={cn(
+            "fade-in-0 absolute inset-0 overflow-hidden transition-opacity duration-300 [&>*]:h-full [&>*]:w-full",
+            className,
+          )}
+        >
+          {props.skeleton || defaultSkeleton}
+        </div>
+      )}
+
+      {error && (
+        <div
+          className={cn(
+            "fade-in-0 absolute inset-0 overflow-hidden transition-opacity duration-300 [&>*]:h-full [&>*]:w-full",
+            className,
+          )}
+        >
+          {props.fallback || defaultFallback}
+        </div>
+      )}
```

Comment:
should these be explicit states? right now it is possible that both render at the same time, no?

Reply:
yeah - I'll refactor this, converted to draft - this is not done yet

Reply:
updated now